arches/app/media/js/views/search.js

```diff
@@ -46,70 +44,45 @@ define([
 
     var CommonSearchViewModel = function() {
         this.filters = {};
-        this.filtersList = _.sortBy(Object.values(SearchComponents), function(filter) {
-            return filter.sortorder;
-        }, this);
+        this.filtersList = Object.values(SearchComponents);
+        this.defaultCoreComponent = this.filtersList.find(component => component.type == "core" && component.config.default == true); // approach: let the backend decide
+        this.coreComponentName = ko.observable(false);
+        this.requiredFiltersLookup = this.filtersList.reduce((lookup, item) => {
+            if (item.config.requiredComponents && item.config.requiredComponents.length > 0) {
+                // Extract the names from requiredComponents and assign them to the lookup object under the component name
+                lookup[item.componentname] = item.config.requiredComponents.map(filter => filter.searchcomponentid);
+            } else {
+                lookup[item.componentname] = [];
+            }
+            return lookup;
+        }, {});
+        this.getRequiredFilters = function(componentname) {
+            // Map these IDs to their corresponding component names from filtersList
+            const requiredComponentNames = this.requiredFiltersLookup[componentname].map(id => {
+                // Find the component in filtersList that matches the id and return its componentname
+                const component = this.filtersList.find(component => component.searchcomponentid == id);
+                return component ? component.componentname : undefined;
+            }).filter(name => name); // This removes any undefined entries in case some IDs don't match any component
+        
+            return requiredComponentNames;
+        };
```

Comment:
Not all search filters (`saved-searches` and `search-export` as examples) extend `BaseFilter` so we'd have to enforce that they do (and change those search filters). Changing how those search components work on the frontend feels like it merits its own ticket though so my vote is to just keep this method in `search.py` since it's pretty generic anyway and can be expected to be called on any search component.